tcl/excel.tcl

```diff
@@ -10,13 +10,25 @@ proc qc::excel_file_create {args} {
     # column_meta: nested dict, eg. {1 {class "foo" width 20} 5 {width 50}}
     # row_meta: nested dict, eg. {0 {class "bar" height 20} 3 {class "foo"}}
     # cell_meta: nested dict, eg. {{5 2} {class "baz" type "string|number|formula|url"} {1 1} {class "bar"}}
-    qc::args2vars $args data formats column_meta row_meta cell_meta timeout
+    qc::args2vars $args data formats column_meta row_meta cell_meta timeout type
     default data {}
     default formats {}
     default column_meta {}
     default row_meta {}
     default cell_meta {}
     default timeout 1000
+    default type "xls"
+    switch $type {
+        "xls" {
+            set writer "Spreadsheet::WriteExcel"
+        }
+        "xlsx" {
+            set writer "Excel::Writer::XLSX"
+        }
+        default {
+            error "Unsupported excel type"
+        }
+    }
 
```

Comment:
Can we make xlsx the default `$type` as part of this pr ?
The xls format may become deprecated in time too.